src/package.cr

```diff
@@ -71,15 +71,27 @@ module Shards
           File.delete(destination)
         end
 
-        begin
-          File.link(source, destination)
-        rescue ex : Errno
-          if {Errno::EPERM, Errno::EXDEV}.includes?(ex.errno)
-            FileUtils.cp(source, destination)
-          else
-            raise ex
+        {% if compare_versions(Crystal::VERSION, "0.34.0-0") > 0 %}
+          begin
+            File.link(source, destination)
+          rescue ex : File::Error
+            if {Errno::EPERM, Errno::EXDEV}.includes?(ex.os_error)
```

Comment:
Can't just catch any `File::Error` and try with `cp`? I mean without checking the `os_error`. I know it's not exactly the same semantics we had before, but if the hardlink fails for any reason try to use a copy.